### ISSUE TITLE
Fix extending rules within overrides

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#5656]
> Clearly describe the bug

When using `extends` within the `overrides` property, the rules don't seem to override.

> Which rule, if any, is the bug related to?

None.

> What code is needed to reproduce the bug?

`input.scss`:

```scss
@mixin {}
```

> What Stylelint configuration is needed to reproduce the bug?

`.stylelintrc.json`:

```json
{
  "rules": {
    "at-rule-no-unknown": true
  },
  "overrides": [
    {
      "files": "**/*.scss",
      "extends": ["./scss.json"]
    }
  ]
}
```

`scss.json`:

```json
{
  "customSyntax": "postcss-scss",
  "rules": {
    "at-rule-no-unknown": null
  }
}
```

> Which version of Stylelint are you using?

14.0.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `npx stylelint input.scss`      

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

N/A

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
input.scss
 1:1  ✖  Unexpected unknown at-rule "@mixin"  at-rule-no-unknown
```

---

What do we think should be the intended behaviour here? I was expecting the rule to be overridden for `scss` files, i.e. to have the following resolved config:

```json
{
  "rules": {
    "at-rule-no-unknown": [
      null
    ]
  },
  "customSyntax": "postcss-scss"
}
```

But `npx stylelint input.scss --print-config` gives me:

```json
{
  "rules": {
    "at-rule-no-unknown": [
      true
    ]
  },
  "customSyntax": "postcss-scss"
}
```

I'm running into this issue while looking into https://github.com/twbs/stylelint-config-twbs-bootstrap/issues/128, where I'd like to write the following:

```json
{
  "extends": ["stylelint-config-standard"],
  "overrides": [
    {
      "files": "**/*.scss",
      "extends": ["stylelint-config-standard-scss"]
    }
  ]
}
```

At-mentioning @ota-meshi because of https://github.com/stylelint/stylelint/pull/5603.

**Comment by ota-meshi** [comment]
@jeddy3 
I'm working on it.
But I think each file will be resolved as follows:

case 1:

```json
{
  "rules": {
    "at-rule-no-unknown": true
  },
  "overrides": [
    {
      "files": "**/*.scss",
      "extends": ["./scss.json"]
    }
  ]
}
```

resolve overrides:

```json
{
  "extends": ["./scss.json"],
  "rules": {
    "at-rule-no-unknown": true
  }
}
```

resolved:

```json
{
  "rules": {
    "at-rule-no-unknown": true
  }
}
```


case 2:

```json
{
  "extends": ["stylelint-config-standard"],
  "overrides": [
    {
      "files": "**/*.scss",
      "extends": ["stylelint-config-standard-scss"]
    }
  ]
}
```

resolve overrides:

```json
{
  "extends": ["stylelint-config-standard", "stylelint-config-standard-scss"],
}
```

resolved:

```json
{
  "rules": {
    // ....
    "at-rule-no-unknown": null
    // ....
  }
}
```

I don't think these are lowering the priority of overrides.
What do you think?

**Comment by jeddy3** [comment]
@ota-meshi Thank you for the clear explanation. 

I had it in my head that `extends` was resolved before `overrides`, but I think you're right. Users would expect a top-level `rules` property to take precedence over any `overrides` define in a config that they extend.
